pkg/entities/set.go

```diff
@@ -40,31 +40,35 @@ const (
 	Undefined = 255
 )
 
-// Do not expose set to IPFIX library user
-// Not creating any interface. Plan to use same struct for Template and Data sets
+type Set interface {
+	CreateNewSet(setType ContentType, templateID uint16) error
+	GetBuffLen() uint16
+	GetSetType() ContentType
+	WriteRecordToSet(recBuffer *[]byte) error
+	FinishSet()
+	AddRecord(elements []*InfoElementValue, templateID uint16, isDecoding bool)
```

Comment:
Can we merge both WriteRecordToSet and AddRecord methods? If isDecoding set, we could skip writing to the buffer as it is not needed, otherwise we should write to buffer.

Reply:
`WriteRecordToSet` is used by exporting process to write records (bytes) to `ep.set`, `AddRecord` is used by users to add `InfoElementWithValue`s as record to set. If combining them, for every set passed into `AddSetAndSendMsg`, we need to copy and add all records to `ep.set`. I would suggest to keep `WriteRecordToSet` and add code in `AddRecord` to write record to set. It will be convenient when sending message. What do you think?

Reply:
As exporting process calls `AddRecord` with false `isDecoding` flag, could skip adding elements to record based on set type and the 'isDecoding' flag? There are 4 scenarios with `setType` and `isDecoding` flag. All these scenarios could be handled in one interface. What about this approach?